File: leetcode/problems/sort_integers_by_the_number_of_1_bits/solution.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        sort(arr.begin(), arr.end(), [](int a, int b) {
          int pa = __builtin_popcount(a);
          int pb = __builtin_popcount(b);
          return pa < pb || (pa == pb && a < b);
        });
        return arr;
    }
};

File: leetcode/problems/sort_integers_by_the_number_of_1_bits/solution.rs
impl Solution {
    pub fn sort_by_bits(mut arr: Vec<i32>) -> Vec<i32> {
        arr.sort_by(|a,b| {
          let pa = a.count_ones();
          let pb = b.count_ones();
          if(pa == pb) {
            return a.cmp(&b);
          }
          return pa.cmp(&pb);
        });
        return arr;
    }
}